src/deferred_data.rs

```diff
@@ -34,6 +36,14 @@ impl<T: DataSourceMut> DeferredDataSourceWrapper<T> {
 }
 
 impl<T: DataSourceMut> DeferredDataSource for DeferredDataSourceWrapper<T> {
+    fn fetch_description(&mut self) {
+
+    }
+
+    fn get_description(&mut self) -> DataSourceDescription {
+        self.data_source.fetch_description()
+    }
+
```

Comment:
Was not sure about this. The other fields are vectors so starting with an empty vector is reasonable. I could add a `description: DataSourceDescription` field here but then it will need some concrete default. Maybe `DataSourceDescription { source_locator: None }` is OK? It seemed a little odd though.

Reply:
Every other method of `DeferredDataSource` is split into two because the response can be delayed (i.e., it may require a network fetch). There are two possible paths here:

 1. Follow the same approach as the other methods, in which case `get_description` MUST return a `Vec`. The rule is that N requests are always mirrored by N responses even if some of the requests are duplicates. Honestly, this seems wasteful to me because it introduces unnecessary complexity for something that (per my understanding) was intended to be an immediate property of the data source.
 2. Make the method immediate. Remove `get` and make `fetch` return `DataSourceDescription`. No need for `Vec`, and no need to store the result in any of the wrappers. This seems vastly simpler.

Reply:
https://github.com/StanfordLegion/prof-viewer/pull/36/commits/c28219e2ce499d54bda9492294abdbf073034ced

👍 2-ish is actually my preference but I was not sure of you wanted to enforce a naming consistency on the API where "fetch" methods on deferred sources did not, themselves, return anything (even if they could in principle)